packages/cli/lib/artifact-generator.js

```diff
@@ -0,0 +1,74 @@
+// Copyright IBM Corp. 2017. All Rights Reserved.
+// Node module: @loopback/cli
+// This file is licensed under the MIT License.
+// License text available at https://opensource.org/licenses/MIT
+
+'use strict';
+const Generator = require('yeoman-generator');
+const utils = require('./utils');
+
+module.exports = class extends Generator {
+  // Note: arguments and options should be defined in the constructor.
+  constructor(args, opts) {
+    super(args, opts);
+    this._setupGenerator();
+  }
+
+  _setupGenerator() {
+    this.artifactInfo.name = this.args[0];
+    this.artifactInfo.defaultName = 'new';
+    this.argument('name', {
+      type: String,
+      required: false,
+      description: 'Name for the ' + this.artifactType,
+    });
+  }
+
+  /**
+   * Override the usage text by replacing `yo loopback4:` with `lb4 `.
+   */
+  usage() {
+    const text = super.usage();
+    return text.replace(/^yo loopback4:/g, 'lb4 ');
+  }
+
+  promptArtifactName() {
+    const prompts = [
+      {
+        type: 'input',
+        name: 'name',
+        message: utils.toClassName(this.artifactInfo.artifactType) + ' name:', // capitalization
+        when: this.artifactInfo.name == undefined,
+        default: utils.toClassName(this.artifactInfo.defaultName),
+        validate: utils.validateClassName,
+      },
+    ];
+
+    return this.prompt(prompts).then(props => {
+      Object.assign(this.artifactInfo, props);
+    });
+  }
+
+  scaffold() {
+    this.destinationRoot(this.artifactInfo.outdir);
```

Comment:
We should not change the `destinationRoot` to a user input as-is. Please do the following:

1. Do not reset `this.destinationRoot`.
2. Use `this.destinationRoot` to check if the project is a LoopBack 4 app based on `this.destinationPath('package.json')`. See http://yeoman.io/authoring/file-system.html.
3. Use `this.destinationPath('src/controllers') to reference the controllers folder.